### PR TITLE
Share bridge progression rules for consistent alerting

### LINKED ISSUE #2355: Extract the shared bridge progression contract
## Problem and scope

Parent layer of #1362. Extract threshold and timestamp evaluation from ui-dashboard/src/lib/bridge-status.ts into a pure @mento-protocol/config export. Keep dashboard presentation and behavior unchanged. Add a machine-readable threshold surface for the later Grafana layer.

## Acceptance
- Preserve strict thresholds: PENDING/SENT >3600s; ATTESTED >900s; QUEUED_INBOUND >86400s. Exclude DELIVERED/CANCELLED/FAILED.
- Preserve ATTESTED lastAttestedTimestamp precedence, then lastUpdatedAt/sentTimestamp/firstSeenAt. Missing, zero and non-finite timestamps remain unknown. Test negative and future values explicitly, destination-first rows and late source upserts.
- Test exact threshold and threshold+1 for every status; preserve dashboard tests. Cross-reference the exported threshold data.
- Run shared-config lint/typecheck/test/build, dashboard lint/typecheck/test/build/size-limit and affected downstream typechecks. Apply stateful-data checklist; no schema or rendered behavior changes.

## Done means
A reviewed parent PR is merged. Exporter and alert rollout remain in #1362. No production action or merge is authorized by this issue.

### PR DESCRIPTION
**Pull request by chapati23**
## The Problem

The dashboard owns the rules that decide when a bridge transfer is stuck. An exporter would otherwise need to copy those rules, which could make alerts disagree with the dashboard.

## The Solution

Move the existing rules into shared-config and let the dashboard use them. The exporter and alert rules can now use the same thresholds and timestamp precedence. Dashboard behavior stays the same; this PR does not enable alerts.

## Details

- Export pure progression timestamp, age and status helpers through `@mento-protocol/config/bridge-status`.
- Publish `bridge-thresholds.json` as the machine-readable threshold source for rule tooling.
- Preserve strict thresholds, terminal exclusion, attestation-first precedence, missing/zero/non-finite fallback, negative epochs and future timestamps. Presentation remains in the dashboard.
- Parent layer of #1362 in [native stack #2370](https://github.com/mento-protocol/monitoring-monorepo/stack/2370): #2358 contract → #2366 exporter → #2369 alerts. Protection base: `main`. Merge from bottom to top only after separate approval for each PR.
- Intended diff and protection base: `main`. No state/schema change or production action.

Closes #2355
Refs #1362

## Validation

- `CI=true pnpm install --frozen-lockfile` passed; manifest additions only expose/package the new modules. This does not publish an npm release.
- Shared-config lint, typecheck, build and 142 tests passed, including 17 progression tests. Coverage passed: 97.82% statements, 96.55% branches, 100% functions, 99.38% lines. These establish tested boundaries and package compilation, not live transfer correctness.
- Dashboard lint, typecheck and all 5,114 unit tests passed. Metrics Bridge and integration-probes typechecks passed. These cover current workspace consumers, not external package users.
- `pnpm dashboard:build` and all five size budgets passed. React Doctor passed 100/100. Chromium visual checks passed 5/5, including the bridge route. Browser coverage exercises the route and runtime import with an empty fixture; numerical status parity is covered by unit tests.
- Trunk formatting, `git diff --check`, and `pnpm docs:index --check` passed. No docs catalog change was needed.
- Scope baseline: six files, 93 non-test added lines; stop threshold 186; base `a746e1c5a45938318303633fa7cddc009f92d0ef`; reviewed and committed tree `bf0b9be2367853ec4d0bca859f736c0533532635`.
- `pnpm agent:closeout-review --base origin/main` refused active nested Codex execution (exit 2). The coordinator independently reviewed all six files against the frozen base/tree and found no defects. This is source review, not a second-model CLI report. Claude Security scan: skipped (Codex surface).

### Published review repair

- Commit `d138336b6d6ab63a2f17ecf65586fe19c4c18485` adds the three bridge artifacts to the publish manifest and removes the JSON dependency from the public declaration.
- The original packed package reproduced TypeScript 5.9.3 error TS1543. An isolated regression now builds and packs the package, checks the tarball, compiles a strict NodeNext consumer with library checking enabled, and verifies its runtime import. It preserves workspace build output. This verifies local distribution behavior; no npm release ran.
- Shared-config build, lint, typecheck and all 143 tests passed. Dashboard production build and all five size budgets passed again. CI contract tests (224), action-pin checks (39 files), workflow trust checks (36 workflows), formatting and diff checks passed.
- The coordinator reviewed repair tree `744c64a6e96fc156c477788826bba35f28203385` against the prior published head with no findings. The closeout command again refused nested Codex execution. The original scope baseline remains unchanged; the final diff has eight files and 96 non-test additions.

## Deferrals

- Exporter implementation and live validation: #2356.
- Alert routing, rules and separately approved production delivery proof: #2357.
- Native stack transition measurements after separately approved merges: #2286.

## Checklist

- [x] The Problem has no more than three bullets.
- [x] The opening explains the old behavior, new behavior, and concrete benefit.
- [x] The opening states any material limit or non-goal that applies.
- [x] A reader can understand the opening without reading the diff.
- [x] Deeper implementation details come after the opening two sections.
- [x] Each Validation claim names its evidence and the nearest stronger claim that evidence does not support.
- [x] Every knowing deferral is listed under Deferrals with a GitHub issue link.
- [x] Architecture decision: existing ADR 0011 governs shared configuration; this extraction adds no new architecture decision.

## Summary by CodeRabbit

- **New Features**
  - Added shared bridge-status logic that identifies in-progress transfers and flags transfers as **STUCK** when they exceed status-specific time thresholds.
  - Improved transfer age calculation using reliable timestamp precedence and fallback handling.
  - Standardized bridge thresholds across supported states.

- **Bug Fixes**
  - Invalid, missing, zero, and non-finite timestamps are now handled safely.
  - Terminal transfers are excluded from stuck-status evaluation.

- **Documentation**
  - Documented bridge progression thresholds and timestamp interpretation rules.

- **Tests**
  - Added coverage for threshold behavior, timestamp handling, and edge cases.

### CODE REVIEW

File: shared-config/package.json [hunk 1]
```diff
@@ -75,6 +76,11 @@
     "./protocol-fee": {
       "types": "./dist/protocol-fee.d.ts",
       "default": "./dist/protocol-fee.js"
+    },
+    "./bridge-thresholds.json": "./bridge-thresholds.json",
+    "./bridge-status": {
+      "types": "./dist/bridge-status.d.ts",
+      "default": "./dist/bridge-status.js"
```
[the remark this answers is not in the document]
Comment (chapati23) [reply shared-config/package.json:83]: Fixed in d138336b6d6ab63a2f17ecf65586fe19c4c18485 — The publish manifest now requires bridge-thresholds.json, dist/bridge-status.js, and dist/bridge-status.d.ts. An isolated npm-pack test also verifies all three artifacts and the packed consumer.

[the remark this answers is not in the document]
Comment (chapati23) [reply shared-config/package.json:83]: Fixed in d138336b6d6ab63a2f17ecf65586fe19c4c18485 — The publish manifest now requires all three bridge artifacts. The packed-consumer regression test checks the actual tarball contents.

File: shared-config/src/bridge-status.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import thresholds from "../bridge-thresholds.json" with { type: "json" };
+
+/** Canonical seconds shared with Grafana rule generation. Queued NTT cannot bypass its 24h inbound window. */
+export const BRIDGE_STUCK_THRESHOLD_SECONDS = thresholds;
+export type BridgeInFlightStatus = keyof typeof thresholds;
```
[the remark this answers is not in the document]
Comment (chapati23) [reply shared-config/src/bridge-status.ts]: Fixed in d138336b6d6ab63a2f17ecf65586fe19c4c18485 — The public status union now derives from BRIDGE_STUCK_THRESHOLD_SECONDS, so its declaration has no JSON import. The packed-package regression compiles a strict NodeNext consumer with skipLibCheck disabled and verifies the runtime import. The original package reproduced TS1543 with TypeScript 5.9.3.

**Comment by chapati23** [comment]
@coderabbitai review

Final source head: `d138336b6d6ab63a2f17ecf65586fe19c4c18485`. Required CI passed. Please review the published packaging repair.
